internal/flock/filesystem_lock_windows.go

```diff
@@ -40,7 +41,12 @@ func Lock(f *os.File) error {
 	if err != nil {
 		return err
 	}
-	defer syscall.CloseHandle(ol.HEvent)
+	defer func() {
+		err := syscall.CloseHandle(ol.HEvent)
+		if err != nil {
+			log.Printf("failed to close handle: %v", err)
```

Comment:
Normally when we use `log` in this codebase we use a log level prefix and include some context about what caused the log line, so maybe...

```suggestion
			log.Printf("[ERROR] failed to close file locking event handle: %v", err)
```

This is not a very specific "context" but probably the best we can do from here since this function doesn't know anything about what's its doing other than that it somehow involves locking. At least it's a relatively distinct string that would come up in code search if someone searched for it. 🤷‍♂️